### ISSUE TITLE
Changes with files_external:notify do not get synced

### ISSUE THREAD

**Issue by dav20011** [nextcloud/server#25408]
### Steps to reproduce:
* Mount an external SMB storage
* Sync the directory with Nextcloud Desktop
* Run the files_external:notify command in verbose mode
* Create a few nested subdirectories in the share (works fine in the root directory of the share)
* Create and modify a file in the subdirectories

### Expected behaviour
* The sync client should pull the updated file from the server

### Actual behaviour
* The notify command reliably shows every change in the share
* The updated file can be viewed in the webinterface
* oc_filecache also contains the new metadata
* Nextcloud Desktop does not sync the file and refuses to do so until the file gets changed via Nextcloud itself (forced sync does not help)
* If notify is not running and Nextcloud updates the data manually (when viewed in the webinterface), sync works just fine

Edit: I investigated a bit further, oc_filecache and oc_filecache_extended show exactly the same changes in both cases, so my guess is that there is another table involved with file sync.

### Server configuration
* OS: Debian Buster
* Web server: nginx
* Database: PostgreSQL
* PHP version: 7.3 (current Debian stable php-fpm)
* Nextcloud version: 20.0.6
* fresh installation (only the necessary things are configured) from nextcloud.com

### Active apps:
Only the standard apps + files_external:
```
  - accessibility: 1.6.0
  - activity: 2.13.4
  - cloud_federation_api: 1.3.0
  - comments: 1.10.0
  - contactsinteraction: 1.1.0
  - dashboard: 7.0.0
  - dav: 1.16.2
  - federatedfilesharing: 1.10.2
  - federation: 1.10.1
  - files: 1.15.0
  - files_external: 1.11.1
  - files_pdfviewer: 2.0.1
  - files_rightclick: 0.17.0
  - files_sharing: 1.12.2
  - files_trashbin: 1.10.1
  - files_versions: 1.13.0
  - files_videoplayer: 1.9.0
  - firstrunwizard: 2.9.0
  - groupfolders: 8.2.0
  - logreader: 2.5.0
  - lookup_server_connector: 1.8.0
  - nextcloud_announcements: 1.9.0
  - notifications: 2.8.0
  - oauth2: 1.8.0
  - password_policy: 1.10.1
  - photos: 1.2.3
  - privacy: 1.4.0
  - provisioning_api: 1.10.0
  - recommendations: 0.8.0
  - serverinfo: 1.10.0
  - settings: 1.2.0
  - sharebymail: 1.10.0
  - support: 1.3.0
  - survey_client: 1.8.0
  - systemtags: 1.10.0
  - text: 3.1.0
  - theming: 1.11.0
  - twofactor_backupcodes: 1.9.0
  - updatenotification: 1.10.0
  - user_status: 1.0.1
  - viewer: 1.4.0
  - weather_status: 1.0.0
  - workflowengine: 2.2.0
```
### Config
```
{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "removed manually"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "pgsql",
        "version": "20.0.6.1",
        "overwrite.cli.url": "http://removed manually",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
    }
}
```
### Client
* Browser: Should not be relevant
* Nextcloud Desktop: 3.1.2
* OS: Windows 10

### Nextcloud log
* Looked it up myself, only shows errors related to the app store (because it was not available for a few hours)

### Nextcloud desktop log
* .owncloudsync.log is empty

**Comment by szaimen** [comment]
As far as I know, is the notify command only marking changed directories as unscanned. The cronjob then will automatically scan those unscanned directories for new files afterwards. So the sync will most likely not happen immediately but should after a new cronjob has run.

**Comment by dav20011** [comment]
I know that, but it does not even sync after I verified that the scan was performed (updated database).

I'll maybe try to use a different method of executing the background tasks, but I don't think that it is related to that.

**Comment by dav20011** [comment]
The issue with Nextcloud not syncing although the metadata was changed in the database seems to be caused by the weird sync system. Files only get updated if their parent directories mtime has been changed and that normally does not happen. I was able to implement a work around PHP script that executes touch on the directory and that fixes the issue, but I think this should be part of the notify handler (if possible over SMB).

Besides that cron still does not do its job, I have to start the scan manually via the scan command. Even manually executing the cron.php does not work.
